zsh completions broken?

Just found this for the canary CI of commit ce4972c07c81a57153b57e4951a20bc44593ca07 but only on macOS 15:
```
_____________________ test_zsh_completion_syntax[archiver] _____________________
[gw0] darwin -- Python 3.14.6 /Users/runner/work/borg/borg/.tox/py314-none/bin/python

archivers = 'archiver'
request = <FixtureRequest for <Function test_zsh_completion_syntax[archiver]>>

    @needs_zsh
    def test_zsh_completion_syntax(archivers, request):
        """Verify the generated Zsh completion script has valid syntax."""
        archiver = request.getfixturevalue(archivers)
        output = cmd(archiver, "completion", "zsh")
        result = _check_shell_syntax(output, "zsh", ".zsh")
>       assert result.returncode == 0, f"Generated Zsh completion has syntax errors: {result.stderr.decode()}"
E       AssertionError: Generated Zsh completion has syntax errors: /var/folders/pd/2_nlvl1s4k121pdk4d5_2c8m0000gn/T/tmpsbi2lge3.zsh:715: parse error near `>'
E         
E       assert 1 == 0
E        +  where 1 = CompletedProcess(args=['zsh', '-n', '/var/folders/pd/2_nlvl1s4k121pdk4d5_2c8m0000gn/T/tmpsbi2lge3.zsh'], returncode=1, stdout=b'', stderr=b"/var/folders/pd/2_nlvl1s4k121pdk4d5_2c8m0000gn/T/tmpsbi2lge3.zsh:715: parse error near `>'\n").returncode

/Users/runner/work/borg/borg/src/borg/testsuite/archiver/completion_cmd_test.py:90: AssertionError

```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,7 @@ dependencies = [
   "platformdirs >=3.0.0, <5.0.0; sys_platform == 'darwin'",  # for macOS: breaking changes in 3.0.0.
   "platformdirs >=2.6.0, <5.0.0; sys_platform != 'darwin'",  # for others: 2.6+ works consistently.
   "argon2-cffi",
-  "shtab>=1.8.0",
+  "shtab>=1.8.0,!=1.8.2,!=1.9.0",  # these generate broken zsh completions, see tqdm/shtab#224
   "backports-zstd; python_version < '3.14'", # for python < 3.14.
   "jsonargparse>=4.47.0",
   "PyYAML>=6.0.2",  # we need to register our types with yaml, jsonargparse uses yaml for config files

